src/Interfaces/AbstractPolyhedron_functions.jl

```diff
@@ -1052,14 +1056,53 @@ julia> project(P, [1, 2]) |> constraints_list
 ```
 """
 function project(P::AbstractPolyhedron{N}, block::AbstractVector{Int}) where {N}
-    if constrained_dimensions(P) ⊆ block
-        clist = [HalfSpace(c.a[block], c.b) for c in constraints_list(P)]
-    else
+    general_case = false
+
+    # cheap case
+    clist = nothing  # allocate later
+    @inbounds for c in constraints(P)
+        subset_status = 0  # initially 0, set to 1 if subset and -1 if disjoint
+        for i in constrained_dimensions(c)
```

Comment:
can we make a small helper function for the loop in line 1065-1082 ? iiuc, this function receives a half-space `c` and a vector of integers `block` and returns two flags, `general_case` and `subset_status`

Reply:
done